Sources/MongoSwift/WriteConcern.swift

```diff
@@ -39,72 +36,67 @@ public class WriteConcern: Codable {
     }
 
     /// Indicates the `W` value for this `WriteConcern`.
-    public var w: W? {
-        let number = mongoc_write_concern_get_w(self._writeConcern)
-        switch number {
-        case MONGOC_WRITE_CONCERN_W_DEFAULT:
-            return nil
-        case MONGOC_WRITE_CONCERN_W_MAJORITY:
-            return .majority
-        case MONGOC_WRITE_CONCERN_W_TAG:
-            if let wTag = mongoc_write_concern_get_wtag(self._writeConcern) {
-                return .tag(String(cString: wTag))
-            }
-        default:
-            break
-        }
-
-        return .number(number)
-    }
+    public var w: W?
 
     /// Indicates whether to wait for the write operation to get committed to the journal.
-    public var journal: Bool? {
-        if mongoc_write_concern_journal_is_set(self._writeConcern) {
-            return mongoc_write_concern_get_journal(self._writeConcern)
-        }
-        return nil
-    }
+    public var journal: Bool?
 
     /// If the write concern is not satisfied within this timeout (in milliseconds),
     /// the operation will return an error. The value MUST be greater than or equal to 0.
-    public var wtimeoutMS: Int64? {
-        let timeout = mongoc_write_concern_get_wtimeout(self._writeConcern)
-        return timeout == 0 ? nil : Int64(timeout)
-    }
+    public var wtimeoutMS: Int64?
 
     /// Indicates whether this is an acknowledged write concern.
     public var isAcknowledged: Bool {
-        return mongoc_write_concern_is_acknowledged(self._writeConcern)
+        if let w = self.w {
+            switch w {
+            case let .number(wNumber):
+                return !((journal == nil || journal == false ) && wNumber == 0)
+            default:
+                return true
+            }
+        }
+        return true
     }
 
     /// Indicates whether this is the default write concern.
     public var isDefault: Bool {
-        return mongoc_write_concern_is_default(self._writeConcern)
+        return self.w == nil && self.journal == nil && self.wtimeoutMS == nil
     }
 
     /// Indicates whether the combination of values set on this `WriteConcern` is valid.
     private var isValid: Bool {
-        return mongoc_write_concern_is_valid(self._writeConcern)
+        if let w = self.w {
```

Comment:
same comment applies about using `if case let` to check the one case we care about